runtime/lib/mirrors_impl.dart

```diff
@@ -1676,4 +1681,18 @@ class _Mirrors {
     }
     return typeMirror;
   }
+
+  static Type _resolveType(Type key, List<Type> typeArguments) {
+    if (!typeArguments.every((_) => _ is Type)) {
```

Comment:
Looks like you're right and I don't have access to `_Type` (also not sure I entirely understand a concept of `patch` used in some places in the SDK and `_Type` is a patch). I might need to figure out how to do this on the C side.